ingestion_tools/scripts/data_validation/fixtures/path.py

```diff
@@ -284,92 +316,202 @@ def canonical_tomo_basenames(
+    instance_seg_annotation_files: List[str],
+    seg_mask_annotation_mrc_files: List[str],
+) -> List[str]:
+    return (
+        point_annotation_files
+        + oriented_point_annotation_files
+        + instance_seg_annotation_files
+        + seg_mask_annotation_mrc_files
+    )
 
-    if len(files) == 0:
-        pytest.skip("No annotation metadata files found.")
 
-    return files
+@pytest.fixture(scope="session")
+def annotation_metadata_files(annotations_dir: str, filesystem: FileSystemApi) -> List[str]:
+    """[Dataset]/[ExperimentRun]/Tomograms/VoxelSpacing[voxel_spacing]/Annotations/[annotation_name].json"""
+    return filesystem.glob(f"{annotations_dir}/*.json")
 
 
-def annotation_files(
+def get_annotation_files_to_metadata_files(
     bucket: str,
-    annotation_metadata: Dict[str, Any],
+    annotation_files: List[str],
+    annotation_metadata: Dict[str, Dict],
     filesystem: FileSystemApi,
     name: str,
     format: Optional[str] = None,
-) -> Dict[str, List[str]]:
-    """[Dataset]/[ExperimentRun]/Tomograms/VoxelSpacing[voxel_spacing]/Annotations/[annotation_name].*"""
-
-    files = {}
+) -> Dict[str, str]:
+    """[Dataset]/[ExperimentRun]/Tomograms/VoxelSpacing[voxel_spacing]/Annotations/[annotation_name].*
+    Helper function for retrieving annotation files and their corresponding metadata files.
+    Fails the test if the annotation file is not found for a given metadata file OR if there are any remaining annotation files.
+    Returns a dictionary of annotation files, annotation_filename -> metadata_filename.
+    """
+
+    remaining_annotation_files = annotation_files.copy()
+    corresponding_annotation_files = {}
     count = 0
 
-    for filename, metadata in annotation_metadata.items():
-        files[filename] = []
-        for annot in metadata["files"]:
-            if annot["shape"] == name and (format is None or annot["format"] == format):
-                file = f"s3://{bucket}/{annot['path']}"
-                if not filesystem.s3fs.exists(file):
-                    pytest.fail(f"{name} {format if format else ''} annotation file not found: {file}")
+    for metadata_filename, metadata in annotation_metadata.items():
+        for annotation in metadata["files"]:
+            if annotation["shape"] == name and (format is None or annotation["format"] == format):
+                file = f"s3://{bucket}/{annotation['path']}"
+                if not filesystem.exists(file):
+                    pytest.fail(f"{name} annotation file not found: {file}")
 
-                files[filename].append(file)
+                corresponding_annotation_files[file] = metadata_filename
+                remaining_annotation_files.remove(file)
                 count += 1
 
+    if len(remaining_annotation_files) > 0:
+        pytest.fail(
+            f"Metadata file not found for {len(remaining_annotation_files)} {name} annotation files.",
+        )
+
     if count == 0:
```

Comment:
nit: Instead of using a counter variable for this, we could check if `corresponding_annotation_files` is not empty.